metadata-ingestion/src/datahub/sql_parsing/sqlglot_lineage.py

```diff
@@ -641,7 +642,7 @@ def _get_direct_raw_col_upstreams(
 
             # Parse the column name out of the node name.
             # Sqlglot calls .sql(), so we have to do the inverse.
-            normalized_col = sqlglot.parse_one(node.name).this.name
+            normalized_col = sqlglot.parse_one(node.name, dialect=dialect).this.name
```

Comment:
can we add a test case that would have failed before but works with this change?

Reply:
I tried to reproduce the issue we talked about in context of the SAP HANA view parsing and I think it wasn't caused by parse_one method: As the identifiers are qualified using the optimize method - and more important with the identify parameter set to true - the name of the node should always be correct, including the capitalization.

For the call to the sql method in the to_node method the dialect is also not provided (https://github.com/tobymao/sqlglot/blob/5df3f5292488df6a8e21abf3b49086c823797e78/sqlglot/lineage.py#L234 and https://github.com/tobymao/sqlglot/blob/5df3f5292488df6a8e21abf3b49086c823797e78/sqlglot/lineage.py#L285), therefore the capitalization should also not be changed by the sql method. Not sure if you meant this when you mentioned the sql method in Slack, but I think you are right - everything is correct.

I have also run some performance tests (using timeit) regarding the dialect instance which would implicitly be created by parse_one and there is basically no difference with and without the dialect instance.